plugins/modules/xen_orchestra.py

```diff
@@ -0,0 +1,312 @@
+    description:
+      - Allows you to create/delete/restart/stop instances on Xen Orchestra
+    options:
+      api_host:
+        description: API host to XOA API.
+        type: str
+      user:
+        description: Xen Orchestra user.
+        required: true
+        type: str
+      password:
+        description: Xen Orchestra password.
+        required: true
+        type: str
+      validate_certs:
+        description: Verify TLS certificate if using HTTPS.
+        type: boolean
+        default: true
+      use_ssl:
+        description: Use wss when connecting to the Xen Orchestra API
+        type: boolean
+        default: true
+      state:
+        description: State in which the Virtual Machine should be
+        choices: ['present', 'started', 'absent', 'stopped', 'restarted']
+        default: present
+      label:
+        description: Label of the Virtual Machine to create, can be used when O(state=present)
+        type: boolean
+        default: false
+      description:
+        description: Description of the Virtual Machine to create, can be used when O(state=present)
+        type: boolean
+        default: false
+      boot_after_create:
+        description: Boot Virtual Machine after creation, can be used when O(state=present)
+        type: boolean
+        default: false
+'''
+
+
+EXAMPLES = r'''
+- name: Create a new virtual machine
+  community.general.xen_orchestra:
+    api_host: xen-orchestra.lab
+    user: user
+    password: passw0rd
+    validate_certs: no
+    state: present
+    template: 355ee47d-ff4c-4924-3db2-fd86ae629676-a3d70e4d-c5ac-4dfb-999b-30a0a7efe546
+    label: This is a test from ansible
+    description: This is a test from ansible
+    boot_after_create: no
+
+- name: Start an existing virtual machine
+  community.general.xen_orchestra:
+    api_host: xen-orchestra.lab
+    user: user
+    password: passw0rd
+    validate_certs: no
+    state: started
+
+- name: Stop an existing virtual machine
+  community.general.xen_orchestra:
+    api_host: xen-orchestra.lab
+    user: user
+    password: passw0rd
+    validate_certs: no
+    state: stop
+
+- name: Restart an existing virtual machine
+  community.general.xen_orchestra:
+    api_host: xen-orchestra.lab
+    user: user
+    password: passw0rd
+    validate_certs: no
+    state: stopped
+
+- name: Delete a virtual machine
+  community.general.xen_orchestra:
+    api_host: xen-orchestra.lab
+    user: user
+    password: passw0rd
+    validate_certs: no
+    state: absent
+'''
+
+import json
+import ssl
+from time import sleep
+
+from ansible.errors import AnsibleError
+
+from ansible_collections.community.general.plugins.module_utils.version import LooseVersion
+from ansible.module_utils.basic import AnsibleModule
+
+# 3rd party imports
+try:
+    HAS_WEBSOCKET = True
+    import websocket
+    from websocket import create_connection
+
+    if LooseVersion(websocket.__version__) <= LooseVersion('1.0.0'):
```

Comment:
So version 1.0.0 is bad, but 1.0.1 is OK? I think you rather mean
```suggestion
    if LooseVersion(websocket.__version__) < LooseVersion('1.0.0'):
```
:)